client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java

```diff
@@ -0,0 +1,93 @@
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.example.criengine.Database.DatabaseWrapper;
+import com.example.criengine.Objects.Book;
+import com.example.criengine.Objects.Profile;
+import com.example.criengine.R;
+import com.google.android.gms.tasks.OnSuccessListener;
+
+public class AddBookActivity extends AppCompatActivity {
+
+    private Profile bookProfile;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_book);
+        // Get database for profile
+        final DatabaseWrapper dbw = DatabaseWrapper.getWrapper();
+
+        // Set buttons and warning field to views
+        final Button cancelButton = findViewById(R.id.newBookCancelButton);
+        final Button saveButton = findViewById(R.id.newBookSaveButton);
+
+        // The fields of the book
+        final EditText bookTitle = findViewById(R.id.newBookTitle);
+        final EditText bookDesc = findViewById(R.id.newBookDesc);
+        final EditText bookAuthor = findViewById(R.id.newBookAuthor);
+        final EditText bookISBN = findViewById(R.id.newBookISBN);
+        final EditText bookImageURL = findViewById(R.id.newBookImageURL);
+        final TextView warning = findViewById(R.id.newBookWarning);
+
+        // Get the current profile
+        dbw.getProfile(dbw.userId).addOnSuccessListener(new OnSuccessListener<Profile>() {
+            @Override
+            public void onSuccess(Profile profile) {
+                bookProfile = profile;
+            }
+        });
+
+        // Save button is clicked
+        saveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!bookTitle.getText().toString().isEmpty() &&
+                        !bookDesc.getText().toString().isEmpty() &&
+                        !bookAuthor.getText().toString().isEmpty() &&
+                        !bookISBN.getText().toString().isEmpty()) {
+
+                    // Create the book
+                    Book newBook = new Book(bookProfile.getUserID(),
+                            bookTitle.getText().toString(),
+                            bookAuthor.getText().toString(),
+                            bookDesc.getText().toString(),
+                            bookISBN.getText().toString(),
+                            "Available");
+
+                    // Adds in image url if present
+                    if (!bookImageURL.getText().toString().isEmpty()) {
+                        newBook.setImageURL(bookImageURL.getText().toString());
+                    }
+
+                    // Adds new book to database
+                    dbw.addBook(newBook);
+                    // Go back to my books
+                    Intent intent = new Intent(v.getContext(), MyBooksActivity.class);
+                    v.getContext().startActivity(intent);
```

Comment:
I think it would be better if the user optionally chooses to return to the other activity. Like if they wanted to add in multiple books they would constantly have to click "add a new book". So instead of forcing them to return to the other activity, have a pop-up notification saying that "Book was Successfully added" (or say it wasn't if it failed) and then keep them on the same activity. Just wipe the info so they can start a new one.

The bottom nav bar should let them go backward/forward if they want.